### ISSUE TITLE
Official Transformer can not get bleu like T2T Transformer

### ISSUE THREAD

**Issue by efeiefei** [tensorflow/models#4478]
### System information
- **What is the top-level directory of the model you are using**: official/transformer
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Centos
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.6
- **Bazel version (if compiling from source)**: No
- **CUDA/cuDNN version**: 9.0
- **GPU model and memory**: V100
- **Exact command to reproduce**: No

### Describe the problem
As official Transformer use embedding_softmax_sharding_weights, we use individual encoder_embedding, decdoer_embedding and softmax.

We use the same input data and params for t2t transfromer and this transformer.
However, we can't get Bleu similar to t2t, always lower than it by 0.5 bleu.
I think their implementation is almost the same
Where is it wrong?

**Comment by robieta** [comment]
I'm afraid I don't follow.
1) Are you using identical weights and hyperparameters?
2) If so, do identical inputs produce identical translations?
3) If so, can you provide an example of text, the translation of said text, the reference translation of said text, and the Bleu scores for said translation produced by the t2t and this transformer.

@k-w-w FYI

**Comment by k-w-w** [comment]
Did you apply the same embedding adjustments to the tensor2tensor implementation?

**Comment by efeiefei** [comment]
@robieta 
We retrain new model with this official transformer, compute its eval bleu, finally compared with the eval bleu of old model trained by t2t transformer. So their weights may be different.
But their training data, eval data, main hyperparameters and total epoch are the same. 

@k-w-w 
We have implemented new embedding and softmax for this transformer, as we didn't use shared_embedding_and_softmax_weights.
I think their basic embedding and weights variable implementation is same, seen from their code. Am I right?

Added, when we trained 1.5x times t2t, we got the similary bleu after avg_checkpoints.
I noticed that, their learning rate and optimizer are the same.
Why is this convergence slow?

Thanks for your reply~

**Comment by k-w-w** [comment]
What commands and environment (number of GPUs and such) did you use to run the t2t transformer and this transformer?

**Comment by efeiefei** [comment]
@k-w-w 
T2T is 4GPU，t2t 1.4.4， tensorflow 1.4
This is 4GPU with horovod (but use one gpu to train 4x step is the similary bleu),  tensorflow 1.6.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by saberkun** [comment]
Closing. The implementation is known to have minor differences: packed inputs, etc. The old estimator transformer is moved as legacy R1 model.
Note that, the checkpoint averaging trick inside transformer paper is not implemented.
